knowledge-base/scheduler-kb-context-menu-on-appointments-and-empty-slots.md

```diff
@@ -0,0 +1,367 @@
+                 @oncontextmenu="@( (MouseEventArgs e) => ShowItemContextMenu(e, Appointment) )">
+                <div style="height:100%" class="k-event-template">@Appointment.Title</div>
+            </div>
+        }
+    </AllDayItemTemplate>
+</TelerikScheduler>
+
+@* ContextMenu for ItemTemplate and AllDayItemTemplate *@
+<TelerikContextMenu @ref="@TheItemContextMenu"
+                    Data="@MenuItems"
+                    TextField="Text"
+                    IconField="Icon"
+                    DisabledField="Disabled"
+                    OnClick="@( async (ContextMenuItem itm) => await MenuClickItemHandler(itm) )">
+</TelerikContextMenu>
+
+@* ContextMenu for SlotTemplate *@
+<TelerikContextMenu @ref="@TheSlotContextMenu"
+                    Data="@MenuSlots"
+                    TextField="Text"
+                    IconField="Icon"
+                    DisabledField="Disabled"
+                    OnClick="@( async (ContextMenuItem itm) => await MenuClickSlotHandler(itm) )">
+</TelerikContextMenu>
+
+<style>
+    .important-appt {
+        color: purple;
+        font-weight: bold;
+        font-size: 1.5em;
+    }
+</style>
+
+@code {
+    private TelerikScheduler<SchedulerAppointment> SchedulerRef { get; set; }
+    private List<SchedulerAppointment> Appointments { get; set; } = new List<SchedulerAppointment>();
+    private DateTime StartDate { get; set; } = DateTime.Today;
+    private SchedulerView CurrView { get; set; } = SchedulerView.Week;
+    private DateTime DayStart { get; set; } = DateTime.Today;
+
+    private SchedulerAppointment LastClickedAppointment { get; set; }
+    private SchedulerSlotTemplateContext LastClickedEmptySlot { get; set; }
+    private TelerikContextMenu<ContextMenuItem> TheItemContextMenu { get; set; }
+    private TelerikContextMenu<ContextMenuItem> TheSlotContextMenu { get; set; }
+
+    private List<ContextMenuItem> MenuItems = new List<ContextMenuItem>()
+        {
+            new ContextMenuItem
+            {
+                Text = "Delete",
+                CommandName = "delete",
+                Icon = SvgIcon.X
+            },
+            new ContextMenuItem
+            {
+                Text = "Toggle Important",
+                CommandName = "toggleimportant",
+                Icon = SvgIcon.Pencil
+            }
+        };
+    private List<ContextMenuItem> MenuSlots = new List<ContextMenuItem>()
+        {
+            new ContextMenuItem
+            {
+                Text = "Create appointment",
+                CommandName = "create",
+                Icon = SvgIcon.File
+            }
+        };
+
+    //handle ContextMenu for Item
+    private async Task ShowItemContextMenu(MouseEventArgs e, SchedulerAppointment appt)
+    {
+        LastClickedAppointment = appt;
+        PrepareMenuItems(LastClickedAppointment);
+        await TheItemContextMenu.ShowAsync(e.ClientX, e.ClientY);
+    }
+
+    private void PrepareMenuItems(SchedulerAppointment appt)
```

Comment:
The method name contains plural, but the prepared item is just one. Why not `DisableItem` ?